text/0000-data-formats.md

```diff
@@ -0,0 +1,151 @@
+# RFC: Data Formats for OpenTelemetry
+
+# Overview
+OpenTelemetry is an observability system, which generates observations, and then sends that data to remote systems for analysis. Describing how this observability data is represented and transmitted is critical part of the OpenTelemetry specification.
+
+**Proposed Layers**
+*   Metadata and Semantic Conventions
+*   Logical Structure
+*   Encodings
+*   Exchange Protocols
+
+
+# Metadata and Semantic Conventions
+
+Spans represent specific operations in and between systems. Some of these operations represent calls that use well-known protocols like HTTP or database calls. Depending on the protocol and the type of operation, additional information is needed to represent and analyze a span correctly in monitoring systems. It is also important to unify how this attribution is made in different languages. This way, the operator will not need to learn specifics of a language and telemetry collected from multi-language micro-service can still be easily correlated and cross-analyzed.
+
+## Requirements
+
+
+### Handle Complex Data
+
+Semantic information is often complex and multi-valued. How we represent semantic data is related to how attributes are defined at the logical layer.
+
+*   At minimum, nested lists of key/value pairs are necessary to represent them.
+*   Is it advantageous to also have strongly typed data structures?
```

Comment:
Not sure if this comment refers to "strongly typed language support" or recorded data to be "strongly typed" (do not allow in java to record Object).

If the data that we support are not typed (strongly typed) there will be vendor issues (locking), an easy scenario is let's assume vendor X supports for Http spans a specific object that describes the operation (HttpDetails). If any instrumentation uses HttpDetails then that means the instrumentation is specific to vendor X not generic. Even though someone may say that HttpDetails will be serialized as a string, this instrumentation offers competitive advantages to vendor X, then other vendors will have to implement that in order to offer the same capability. Also this becomes even worse if the vendor X controls the definition of the HttpDetails.

If you refer about languages that they do not have a strongly type system then we should do our best (using static code checks or other mechanisms) to ensure only accepted types are used.

So I suggest that we do not support random objects in order to discourage any vendor locking and to try to offer vendor neutrality.